### PR TITLE
Add API functions and completions for plugins

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: api/sys_plugins.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	"fmt"
+	"net/http"
+)
+
+// ListPluginsInput is used as input to the ListPlugins function.
+type ListPluginsInput struct{}
+
+// ListPlugins lists all plugins in the catalog and returns their names as a
+// list of strings.
+func (c *Sys) ListPlugins(i *ListPluginsInput) ([]string, error) {
```
Comment (jefferai) [inline api/sys_plugins.go]: Please make this a ListPluginsResponse (or ListPluginsOutput) rather than []string.

Comment (sethvargo) [reply api/sys_plugins.go]: Updated

Comment (sethvargo) [reply api/sys_plugins.go]: And added tests

**Comment by jefferai** [comment]
LGTM, asking for another.

**Comment by jefferai** [comment]
Thanks!

**Pull request merged**